### PR TITLE
fix: package version. now stored in packdir in sfdx project

### PR DESCRIPTION
**Pull request by peternhale**
@W-11489082@

### CODE REVIEW

File: src/package/packageVersion.ts [hunk 1]
```diff
@@ -217,4 +231,78 @@ export class PackageVersion {
...
+          tooling: true,
+        });
+        if (pkgQueryResult.IsOrgDependent) {
+          w = Duration.seconds((60 / pollInterval.seconds) * wait.seconds);
+        }
+      } catch {
+        // do nothing
+      }
+    }
+    return w;
```
Comment (mshanemc) [inline src/package/packageVersion.ts:263]: the let-free version
```suggestion
    // If we are polling check to see if the package is Org-Dependent, if so, update the poll time
    if (wait.milliseconds > 0) {
      const query = `SELECT IsOrgDependent FROM Package2 WHERE Id = '${resolvedPackageId}'`;
      try {
        const pkgQueryResult = await this.connection.singleRecordQuery<PackagingSObjects.Package2>(query, {
          tooling: true,
        });
        if (pkgQueryResult.IsOrgDependent) {
          return Duration.seconds((60 / pollInterval.seconds) * wait.seconds);
        }
      } catch {
        // do nothing
      }
    }
    return wait;
```

File: src/package/packageVersion.ts [hunk 2]
```diff
@@ -217,4 +231,78 @@ export class PackageVersion {
...
     return updateResult;
   }
+
+  /**
+   * Increase the wait time for a package version that is org dependent.
+   *
+   * @param resolvedPackageId
+   * @param pollInterval
+   * @param wait
+   * @private
```
Comment (mshanemc) [inline src/package/packageVersion.ts:241]: these don't add anything beyond what TS is gonna provide.

File: src/package/packageVersion.ts [hunk 3]
```diff
@@ -217,4 +231,78 @@ export class PackageVersion {
...
+    }
+  }
+
+  private async generatePackageDirectory(
+    packageVersion: PackagingSObjects.Package2Version,
+    withProject: SfProject,
+    packageVersionVersionString: string
+  ) {
+    const pkg = await (await Package.create({ connection: this.connection })).getPackage(packageVersion.Package2Id);
+    const pkgDir =
```
Comment (mshanemc) [inline src/package/packageVersion.ts:296]: putting some appropriate overloads on the utils for this class might help TS know the return type (should match the T of the first argument or be undefined).

I'm not a fan of `{} as NamedPackageDir` (it's just not).

If `getConfigPackageDirectory` returns an undefined, this method should handle that.

File: src/utils/packageUtils.ts [hunk 4]
```diff
@@ -942,9 +942,8 @@ async function _generatePackageAliasEntry(
   let packageName;
   if (!aliasForPackageId || aliasForPackageId.length === 0) {
     const query = `SELECT Name FROM Package2 WHERE Id = '${packageId}'`;
-    packageName = await connection.tooling
-      .query<PackagingSObjects.Package2>(query)
-      .then((pkgQueryResult) => pkgQueryResult.records[0]?.Name);
+    const package2 = await connection.singleRecordQuery<PackagingSObjects.Package2>(query, { tooling: true });
```
Comment (mshanemc) [inline src/utils/packageUtils.ts:945]: +1

File: src/package/packageVersion.ts [hunk 5]
```diff
@@ -217,4 +231,78 @@ export class PackageVersion {
...
+      getConfigPackageDirectory(withProject.getPackageDirectories(), 'id', pkg.Id) ?? ({} as NamedPackageDir);
+    pkgDir.versionNumber = packageVersionVersionString;
+    pkgDir.versionDescription = packageVersion.Description;
+    const packageDirs = withProject.getPackageDirectories().map((pd) => {
+      if (pkgDir['id'] === pd['id']) {
+        return pkgDir;
+      }
+      return pd;
+    });
```
Comment (mshanemc) [inline src/package/packageVersion.ts:305]: if the types are correct, there should be an id, right?
```suggestion
    const packageDirs = withProject.getPackageDirectories().map((pd) => pkgDir.id === pd.id ? pkgDir : pd)
```

**Review by mshanemc — APPROVED** [review]
no objections, just ideas

File: src/package/packageVersion.ts (hunk 5 above)
Comment (peternhale) [reply src/package/packageVersion.ts:304]: @mshanemc Actually `id` is not a property of a package dir according to the formal schema for sfdx-project.json. This property is injected during package version create. I will check with packaging team as to why it is needed and should it be part of the formal schema.

**Review by WillieRuemmele — APPROVED** [review]

**Pull request merged**